queue/group_remote_mongo_single.go

```diff
@@ -10,40 +10,46 @@ import (
 	"github.com/mongodb/grip/recovery"
 	"github.com/pkg/errors"
 	"go.mongodb.org/mongo-driver/bson"
-	"go.mongodb.org/mongo-driver/mongo"
 )
 
+// remoteMongoQueueGroupSingle is a group of queues backed by MongoDB that are
+// multiplexed into a single MongoDB collection and isolated by group
+// namespaces.
 type remoteMongoQueueGroupSingle struct {
 	canceler context.CancelFunc
-	client   *mongo.Client
 	opts     MongoDBQueueGroupOptions
-	dbOpts   MongoDBOptions
 	cache    GroupCache
 }
 
-// NewMongoDBSingleQueueGroup constructs a new remote queue group. If ttl is 0, the queues will not be
-// TTLed except when the client explicitly calls Prune.
-func NewMongoDBSingleQueueGroup(ctx context.Context, opts MongoDBQueueGroupOptions, client *mongo.Client, mdbopts MongoDBOptions) (amboy.QueueGroup, error) {
+// NewMongoDBSingleQueueGroup constructs a new remote queue group. If the TTL is
+// 0, the queues will not be TTLed except when the client explicitly calls
+// Prune.
+//
+// The MongoDB single remote queue group multiplexes all queues into a single
+// collection. Group name as a means to namespace each queue within the
```

Comment:
`Group name is used as a means...`

Reply:
Great catch! I removed this `GroupName` comment since it's not actually pertinent to the constructor's caller. I might eventually make `UseGroups` and `GroupName` unexported since they should really only be set by the queue groups and not by end users.